### PR TITLE
refactor: Make seeding config more understandable 

### PR DESCRIPTION
**Pull request by LuisFelipeCoelho**
Re-writing mostly all comments from seeding configuration to make them more understandable.

### CODE REVIEW

File: Core/include/Acts/Seeding/SeedFinderOrthogonalConfig.hpp
Comment (paulgessinger) [inline Core/include/Acts/Seeding/SeedFinderOrthogonalConfig.hpp:1]: If these become doxygen comments, we could pull most or all of this into the main documentation.

File: Core/include/Acts/Seeding/SeedFinderConfig.hpp
[the remark this answers is not in the document]
Comment (LuisFelipeCoelho) [reply Core/include/Acts/Seeding/SeedFinderConfig.hpp:1]: reordering + adding and changing comments. But no configuration is renamed or removed

File: Core/include/Acts/Seeding/SeedConfirmationRangeConfig.hpp
[the remark this answers is not in the document]
Comment (LuisFelipeCoelho) [reply Core/include/Acts/Seeding/SeedConfirmationRangeConfig.hpp:1]: Sure I will do that

File: Core/include/Acts/Seeding/SeedFinderConfig.hpp [hunk 1]
```diff
@@ -25,111 +25,147 @@ template <typename SpacePoint>
...
 
-  // seed confirmation
-  bool seedConfirmation = false;
-  // parameters for central seed confirmation
-  SeedConfirmationRangeConfig centralSeedConfirmationRange;
-  // parameters for forward seed confirmation
-  SeedConfirmationRangeConfig forwardSeedConfirmationRange;
+  // Maximum allowed cotTheta between two space-points in doublet, used to check
+  // if forward angle is within bounds
+  float cotThetaMax = 10.01788;  // equivalent to 3 eta (pseudorapidity)
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/SeedFinderConfig.hpp:111]:
```suggestion
  float cotThetaMax = 10.01788;  // equivalent to eta = 3 (pseudorapidity)
```
(Only if possible without linebreak)

File: Core/include/Acts/Seeding/SeedFinderConfig.hpp [hunk 2]
```diff
@@ -25,111 +25,147 @@ template <typename SpacePoint>
...
+  // of triplets with maximum multiple scattering effect (produced by the
+  // minimum allowed pT particle) + a certain uncertainty term. Check the
+  // documentation for more information
+  // https://acts.readthedocs.io/en/latest/core/reconstruction/pattern_recognition/seeding.html
+  float minPt = 400. * Acts::UnitConstants::MeV;
+  // Number of sigmas of scattering angle to be considered in the minimum pT
+  // scattering term
   float sigmaScattering = 5;
-  // Upper pt limit for scattering calculation
+  // Term that accounts for the thickness of scattering medium in radiation lengths in the Lynch \& Dahl correction to the Highland equation
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/SeedFinderConfig.hpp:135]:
```suggestion
  // Term that accounts for the thickness of scattering medium in radiation lengths in the Lynch & Dahl correction to the Highland equation
```

File: Core/include/Acts/Seeding/SeedFinderOrthogonalConfig.hpp [hunk 3]
```diff
@@ -64,42 +38,113 @@ struct SeedFinderOrthogonalConfig {
...
+  // Minimum transverse momentum (pT) used to check the r-z slope compatibility
+  // of triplets with maximum multiple scattering effect (produced by the
+  // minimum allowed pT particle) + a certain uncertainty term. Check the
+  // documentation for more information
+  // https://acts.readthedocs.io/en/latest/core/reconstruction/pattern_recognition/seeding.html
+  float minPt = 400. * Acts::UnitConstants::MeV;
+  // Number of sigmas of scattering angle to be considered in the minimum pT
+  // scattering term
+  float sigmaScattering = 5;
+  // Term that accounts for the thickness of scattering medium in radiation lengths in the Lynch \& Dahl correction to the Highland equation
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/SeedFinderOrthogonalConfig.hpp:115]:
```suggestion
  // Term that accounts for the thickness of scattering medium in radiation lengths in the Lynch & Dahl correction to the Highland equation
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]

File: Core/include/Acts/Seeding/SeedFinderConfig.hpp (hunk 2 above)
Comment (LuisFelipeCoelho) [reply Core/include/Acts/Seeding/SeedFinderConfig.hpp:135]: Thanks for the fixes @AJPfleger

**Comment by paulgessinger** [comment]
Do we have an idea why the physmon output changes?

**Comment by LuisFelipeCoelho** [comment]
> Do we have an idea why the physmon output changes?

Maybe some default values of the orthogonal seeding were different from the default, I will check if that is the case

**Comment by LuisFelipeCoelho** [comment]
FYI: @CarloVarni

File: Core/include/Acts/Seeding/SeedConfirmationRangeConfig.hpp [hunk 4]
```diff
@@ -15,48 +15,50 @@
...
   std::size_t nTopForLargeR = 0;
-  // Minimum number of compatible outer space-points required in quality seed
-  // confirmation if inner space-points radius is smaller than rMaxSeedConf
+  /// Minimum number of compatible outer space-points required in quality seed
+  /// confirmation if inner space-points radius is smaller than rMaxSeedConf
   std::size_t nTopForSmallR = 0;
 
-  // Minimum radius for inner seed component required in quality in seed
-  // confirmation
+  /// Minimum radius for inner seed component required in quality in seed
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/SeedConfirmationRangeConfig.hpp:55]:
```suggestion
  /// Minimum radius for inner seed component required in quality seed
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [c561d529ea1ac3cd660b5cf7f505f69b8b628ddb]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34464351) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6572094)!

**Review by paulgessinger — CHANGES_REQUESTED** [review]
~~This seems to fail the Athena build. Can you either make it so it's backward compatible @LuisFelipeCoelho or we need to hold this until v32 next week. ~~

Build failure is likely because the branch is not even with main. I'll update and retrigger

**Comment by paulgessinger** [comment]
Ok it's green now, so good to go.

**Review by paulgessinger — APPROVED** [review]

**Review by paulgessinger — APPROVED** [review]

**Pull request merged**